discovery/azure/azure.go

```diff
@@ -608,7 +614,13 @@ func mapFromVM(vm armcompute.VirtualMachine) virtualMachine {
 }
 
 func mapFromVMScaleSetVM(vm armcompute.VirtualMachineScaleSetVM, scaleSetName string) virtualMachine {
-	osType := string(*vm.Properties.StorageProfile.OSDisk.OSType)
+	var osType string
+	if vm.Properties != nil &&
```

Comment:
Move this inside the existing `if vm.Properties != nil` (previously on line 621)